services/src/main/java/org/exoplatform/gamification/rest/UserReputationEndpoint.java

```diff
@@ -0,0 +1,291 @@
+import org.exoplatform.gamification.service.dto.configuration.BadgeDTO;
+import org.exoplatform.gamification.service.effective.GamificationService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.service.rest.Util;
+import org.exoplatform.social.webui.Utils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Path("/gamification/reputation")
+@Produces(MediaType.APPLICATION_JSON)
+public class UserReputationEndpoint implements ResourceContainer {
+
+    private static final Log LOG = ExoLogger.getLogger(UserReputationEndpoint.class);
+
+    private final CacheControl cacheControl;
+
+    protected GamificationService gamificationService = null;
+
+    protected IdentityManager identityManager = null;
+
+    protected BadgeService badgeService = null;
+
+    public UserReputationEndpoint() {
+
+        this.cacheControl = new CacheControl();
+
+        cacheControl.setNoCache(true);
+
+        cacheControl.setNoStore(true);
+
+        gamificationService = CommonsUtils.getService(GamificationService.class);
+
+        identityManager = CommonsUtils.getService(IdentityManager.class);
+
+        badgeService = CommonsUtils.getService(BadgeService.class);
+    }
+
+    @GET
+    @RolesAllowed("users")
+    @Path("point/status")
+    public Response getReputationStatus(@Context UriInfo uriInfo, @Context HttpServletRequest request) {
+
+        ConversationState conversationState = ConversationState.getCurrent();
+
+        if (conversationState != null) {
+            try {
+
+                // Compute user id
+                String actorId = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, conversationState.getIdentity().getUserId(), false).getId();
+
+                JSONObject reputation = new JSONObject();
+                reputation.put("points", gamificationService.getUserGlobalScore(actorId));
+
+
+                return Response.ok().cacheControl(cacheControl).entity(reputation.toString()).build();
+
+            } catch (Exception e) {
+
+                LOG.error("Error listing all rules ", e);
+
+                return Response.serverError()
+                        .cacheControl(cacheControl)
+                        .entity("Error listing all rules")
```

Comment:
it's not about listing the rules, the same for the error log before

Reply:
Done